src/components/search.js

```diff
@@ -41,50 +41,43 @@ class Search extends Emitter {
         this.config.includeTimetoken = true;
         this.config.stringifiedTimeToken = true;
         this.config.count = this.config.count || 100;
-
         this.config.pages = this.config.pages || 10;
 
-        this.needleCount = 0;
-
-        this.firstTT = 0;
-        this.lastTT = 0;
+        /** @private */
+        this.maxPage = this.config.pages;
+        /** @private */
+        this.numPage = 0;
 
-        this.firstPage = true;
+        /** @private */
+        this.referenceDate = this.config.end || 0;
 
         /**
-        * @private
-        */
-        this.sortHistory = (messages, desc) => {
-
-            messages.sort((a, b) => {
-                let e1 = desc ? b : a;
-                let e2 = desc ? a : b;
-                return parseInt(e1.timetoken, 10) - parseInt(e2.timetoken, 10);
-            });
-
-            return messages;
-
-        };
+         * Flag which represent whether there is potentially more data available in {@link Chat} history. This flag can
+         * be used for conditional call of {@link Chat#search}.
+         * @type {boolean}
+         */
+        this.hasMore = true;
+        /** @private */
+        this.messagesBetweenTimetokens = this.config.start > '0' && this.config.end > '0';
+        /** @private */
+        this.needleCount = 0;
 
         /**
          * Call PubNub history in a loop.
-         * Unapologetically stolen from https://www.pubnub.com/docs/web-javascript/storage-and-history
          * @private
          */
         this.page = (pageDone) => {
+            let searchConfiguration = Object.assign({}, this.config, { start: this.referenceDate });
```

Comment:
```this.config.end === history({start})```. Confusing, but I get it.

Reply:
@ianjennings the reason why I use separate variable, is not to modify configuration, which has been passed by user.
According to interface `config` is public variable and user may decide to read from it, what he used for search. He will be very surprised to find different dates in it.

Reply:
@ianjennings the reason why I use separate variable, is not to modify configuration, which has been passed by user.
According to interface config is public variable and user may decide to read from it, what he used for search. He will be very surprised to find different dates in it.